postgis: update mapper dynamically

pjonsson opened the pull request:
### Reason for this pull request

Inherit SpatialIdx<epsg> from the Base
class, so the mapper is managed
automatically by Base.

### Proposed changes

- 
- 



 - [ ] Closes #xxxx
 - [ ] Tests added / passed
 - [X] Fully documented, including `docs/about/whats_new.rst` for all changes

<!--
See https://github.com/blog/2111-issue-and-pull-request-templates for more
information on pull request templates.

-->


<!-- readthedocs-preview opendatacube start -->
----
📚 Documentation preview 📚: https://opendatacube--1831.org.readthedocs.build/en/1831/

<!-- readthedocs-preview opendatacube end -->

pjonsson committed:
postgis: update mapper dynamically

Inherit SpatialIdx<epsg> from the Base
class, so the mapper is managed
automatically by Base.

pjonsson committed:
spatial: use dict literal

The previous line defines a dict
literal, so move the definition
of the extent column into the dict
literal.

pjonsson commented:
I believe this PR is a small step towards being able to use the ORM, but there are quite a few things that still confuse me, for example why there was a list of tables in the first place. According to CodeCov there is coverage for the changed lines though, so considering the tests pass I might have been lucky with my edits.

SpacemanPaul commented:
> I believe this PR is a small step towards being able to use the ORM, but there are quite a few things that still confuse me, for example why there was a list of tables in the first place.

IIRC the intention was mostly to keep the dynamically created spatial index tables (that vary from deployment to deployment) separate from the statically created other tables, particularly with respect to keeping the spatial index tables outside of Alembic's view.

pjonsson committed:
postgis: comment on ALL_STATIC_TABLES

pjonsson commented:
> [..] particularly with respect to keeping the spatial index tables outside of Alembic's view.

I will assume there is no testing of this kind of stuff in the test suite, so I dropped the patch that removes the list, and added a comment about why it's there.
